### PR TITLE
Dyninst: add standalone test

### PR DESCRIPTION
**Pull request by hainest**
It runs correctly after applying #37841.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dyninst/package.py [hunk 1]
```diff
@@ -183,3 +183,8 @@ def cmake_args(self):
                 args.append("-DENABLE_STATIC_LIBS=NO")
 
         return args
+
+    def test_ptls(self):
+      parseThat = which(join_path(self.prefix.bin, "parseThat"))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dyninst/package.py]: Mind adding a docstring here for the test description?

Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dyninst/package.py]: Optional. You don't have to use `join_path` here to add `parseThat` to the path.

```suggestion
      parseThat = which(self.prefix.bin.parseThat)
```

File: var/spack/repos/builtin/packages/dyninst/package.py [hunk 2]
```diff
@@ -183,3 +183,8 @@ def cmake_args(self):
...
+      os.environ["DYNINSTAPI_RT_LIB"] = join_path(self.prefix, "lib", "libdyninstAPI_RT.so")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dyninst/package.py]: Optional. Same here for joining `self.prefix` and `lib`, though we don't want an `os.sep` to replace the `.` in the library name.

```suggestion
      os.environ["DYNINSTAPI_RT_LIB"] = join_path(self.prefix.lib, "libdyninstAPI_RT.so")
```

**Review by tldahlgren — DISMISSED** [review]
Works for me.

I have some minor suggestions.  Let me know if you want to add any of them or just want the PR merged.

**Comment by tldahlgren** [comment]
Looks like there are style issues from `black` and `flake`.

File: var/spack/repos/builtin/packages/dyninst/package.py (hunk 1 above)
Comment (hainest) [reply var/spack/repos/builtin/packages/dyninst/package.py]: agreed.

Comment (hainest) [reply var/spack/repos/builtin/packages/dyninst/package.py]: Oh, I didn't realize that the path objects did that. Thanks for the heads-up!

**Comment by hainest** [comment]
@tldahlgren I'm good with this. Feel free to merge when you can.

File: var/spack/repos/builtin/packages/dyninst/package.py [hunk 3]
```diff
@@ -183,3 +183,14 @@ def cmake_args(self):
                 args.append("-DENABLE_STATIC_LIBS=NO")
 
         return args
+
+    def test_ptls(self):
+        """
+        Run parseThat on /bin/ls to rewrite with basic instrumentation
+        """
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dyninst/package.py]: Thanks for the changes.  Formatting the docstring this way yields:
```
==> Spack test dyninst
==> Testing package dyninst-12.3.0-qp5gcub
==> [2023-05-24-15:11:53.660958] test: test_ptls: 
        Run parseThat on /bin/ls to rewrite with basic instrumentation
        
==> [2023-05-24-15:11:53.676657] '/usr/WS1/dahlgren/releases/spack/opt/spack/linux-rhel8-broadwell/gcc-10.3.1/dyninst-12.3.0-qp5gcub7y4svxcphngown7gp3inafdik/bin/parseThat' '--binary-edit=/g/g21/dahlgren/.spack/test/j7dex34frx7y5qwev4v3rno2k4pqrney/ls.rewritten' '/bin/ls'
...
```

Would you mind tightening it up a little?

```suggestion
        """Run parseThat on /bin/ls to rewrite with basic instrumentation"""
```

Comment (hainest) [reply var/spack/repos/builtin/packages/dyninst/package.py]: Fixed.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/dyninst/package.py]: Thanks!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**